### DISCUSSION TITLE
Create a plugin for IntelliJ IDEA for Pi4J projects

### DISCUSSION THREAD

**Discussion by FDelporte** [Pi4J/pi4j#111] [Ideas]
One of the goals of the Pi4J project is to document and simplify the "getting started" process of Java development on the Raspberry Pi. As such it could be a good idea to create an IntelliJ IDEA plugin to assist in this process.

Students of the FHNW project have already done some work in this direction by creating run configurations in the CrowPi project to deploy code from developer PC to the Raspberry Pi and run with remote debugging: https://github.com/Pi4J/pi4j-example-crowpi/tree/main/.run

Some of the goals of this plugin could be to:
* Init a new Java (minimal) Maven project with Pi4J
* The same + JavaFX
* Add run/debug configurations for remote debug
* Help to identify GPIO numbers and usages
* etc.

How this plugin needs to be created is not clear for me yet, I just created a repository based on the template provided by JetBrains: https://github.com/Pi4J/pi4j-intellij-plugin

Anyone who has ideas on what such a plugin could provide as features + how to implement, please join this discussion! :-)

**Comment by bigalnz** [comment]
+1 on this after 8 hours trying to get the archetype loaded without success in Intellij.
